plugins/catalog-backend/src/ingestion/processors/ApiDefinitionAnnotationProcessor.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ComponentEntity,
+  Entity,
+  LocationSpec,
+} from '@backstage/catalog-model';
+import { LocationProcessor, LocationProcessorEmit } from './types';
+
+const API_DEFINITION_TYPE_ANNOTATION = /^backstage.io\/api-definition-(.+)$/;
+
+export function extractApiDefinitionTypeAnnotations(entity: Entity): string[] {
+  if (!entity.metadata.annotations) {
+    return [];
+  }
+
+  return Object.keys(entity.metadata.annotations).filter(a =>
+    a.match(API_DEFINITION_TYPE_ANNOTATION),
+  );
+}
+
+export function extractApiDefinitionType(annotationName: string): string {
+  return annotationName.match(API_DEFINITION_TYPE_ANNOTATION)![1];
+}
```

Comment:
These two could mayhap be a single `getApiDefinitionUrls(entity: Entity): Record<string, URL>`, returning something like

```js
{
  openapi: new URL('https://example.com/openapi.yaml')
}
```

where the URL is always absolute. If the input annotation was relative, you'd make that into an absolute one relative to the location that the entity itself is located on.